bouncer/src/repo/core/handler/fileservice/repo_file_handler_fs.cpp

```diff
@@ -133,15 +133,37 @@ std::string FSFileHandler::uploadFile(
 	path /= keyName;
 	ss << keyName;
 	int retries = 0;
-	bool failed;
+	bool failed = false;
 	do {
 		std::ofstream outs(path.string(), std::ios::out | std::ios::binary);
-		outs.write((char*)bin.data(), bin.size());
-		outs.close();
-		if (failed = (!outs || !repo::lib::doesFileExist(path))) {
-			repoError << "Failed to write to file " << path.string() << ((retries + 1) < 3 ? ". Retrying... " : "");
+		if (outs.good()) 
+		{
+			outs.write((char*)bin.data(), bin.size());
+			outs.close();
+		}
+		else if(outs.bad() || outs.fail()) 
+		{
+			failed = true;
+			if (repo::lib::doesFileExist(path)) {
```

Comment:
I don't think this is an issue? if we're overwriting an existing file we are overwriting an existing file. it wouldn't be an error?

Reply:
The above code is reverted back as per the discussion.